src/Donation/DonationRequestUtils.php

```diff
@@ -88,6 +91,11 @@ public function createFromRequest(Request $request, ?Adherent $currentUser): Don
         return $donation;
     }
 
+    public function getDuration(Request $request): int
+    {
+        return (int) $request->query->getInt('abonnement', PayboxPaymentSubscription::NONE);
```

Comment:
Cast is useless, getInt returns only integer value

Reply:
Correct I did it without thinking :s